src/crystal/system/unix/file.cr

```diff
@@ -252,14 +252,29 @@ module Crystal::System::File
     flock LibC::FlockOp::UN
   end
 
-  private def flock(op : LibC::FlockOp, blocking : Bool = true)
-    op |= LibC::FlockOp::NB unless blocking
+  private def flock(op : LibC::FlockOp, retry : Bool) : Nil
+    op |= LibC::FlockOp::NB
 
-    if LibC.flock(fd, op) != 0
-      raise IO::Error.from_errno("Error applying or removing file lock")
+    if retry
+      until flock(op)
+        ::Fiber.yield
```

Comment:
If no other fibers have work to be done, what prevents this from spinning and using up 100% of the CPU until the file lock is obtained?

Reply:
Fair point. A `sleep 5.milliseconds` will significantly lower the pressure on the CPU in such scenario without blocking for too long.

Reply:
Yeah that's definitely not ideal. This is just a basic first step and we're missing other features like a timeout (as suggested in https://github.com/crystal-lang/crystal/issues/12721#issuecomment-1308276097). Timeout and sleep are related, so might make sense to consider that together.

Reply:
Do we know when the timeout feature will be available in the crystal release for this function?

Reply:
Actually, I'm not sure about the timeout. Maybe that should rather be handled more generically.

We should definitely sleep the fiber, though. I've posted a patch for that in #12861.